src/automata/RequestResponseRequester.cpp

```diff
@@ -110,31 +104,35 @@ void RequestResponseRequester::onNextFrame(Frame_ERROR&& frame) {
 }
 
 void RequestResponseRequester::onNextFrame(Frame_RESPONSE&& frame) {
-  bool end = false;
   switch (state_) {
     case State::NEW:
       // Cannot receive a frame before sending the initial request.
       CHECK(false);
```

Comment:
yes, if end up failing in the CHECK, we broke the invariant and didn't handle some cases. The frames sent over the wire should never cause the check fail.